src/utils/helper.ts

```diff
@@ -80,6 +97,18 @@ export function formatDecimalUnits(value: BigDecimal, decimals: BigInt): BigDeci
 	return value.div(scaler.toBigDecimal());
 }
 
+
+export function calcWeightedAverage(values: BigInt[], weights: BigInt[]): BigInt {
+
+	let numerator = BigInt.fromI32(0);
+	let denominator = BigInt.fromI32(0);
+	for (let i = 0; i < values.length; i++) {
+		denominator = denominator.plus(values[i].times(weights[i]))
+		numerator = numerator.plus(values[i]);
+	}
+	return numerator.div(denominator)
```

Comment:
pretty sure we will need to multiply numerator here by `1 * 10 ** decimals` before dividing